src/read/macho/dyld_cache.rs

```diff
@@ -31,8 +31,24 @@ where
     mappings: &'data [macho::DyldCacheMappingInfo<E>],
 }
 
-// This is the offset of the images_across_all_subcaches_count field.
-const MIN_HEADER_SIZE_SUBCACHES: u32 = 0x1c4;
+/// A slice of structs describing each subcache. The struct gained
+/// an additional field (the file suffix) in dyld-1042.1 (macOS 13 / iOS 16),
+/// so this is an enum of the two possible slice types.
+#[derive(Debug, Clone, Copy)]
+pub enum DyldSubCacheSlice<'data, E: Endian> {
```

Comment:
Would it be a pain to make this non_exhaustive?

Reply:
I'll make that change, I don't think it'll cause me a lot of grief.